digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala

```diff
@@ -2306,6 +2307,56 @@ object DataFixture {
     println("Complete at time: " + DateTime.now())
   }
 
+  def fillNewRoadLinksWithPreviousInfo(): Unit = {
+    println("\nStart process to fill new road links with previous data, only for change type 12")
+    println(DateTime.now())
+    println("\n")
+
+    //Get All Municipalities
+    val municipalities: Seq[Int] = OracleDatabase.withDynSession {
+      Queries.getMunicipalities
+    }
+
+    OracleDatabase.withDynTransaction {
+      municipalities.foreach { municipality =>
+        println("\nWorking at Municipailty: " + municipality)
+        val (roadLinks, changes) = roadLinkService.getRoadLinksWithComplementaryAndChangesFromVVHByMunicipality(municipality, newTransaction = false)
+        val filteredRoadLinks = roadLinks.filter(r => r.isCarRoadOrCyclePedestrianPath)
+        val changesToTreat = changes.filter(c => c.changeType == New.value && filteredRoadLinks.exists(_.linkId == c.newId.getOrElse(0)))
+        val speedLimitsAlreadyExistents = changesToTreat.flatMap { ctt =>
+          speedLimitService.getExistingAssetByRoadLink(filteredRoadLinks.find(_.linkId == ctt.newId.getOrElse(0)).get, false)
+        }
+        val changesWithoutSpeedLimitCreated = changesToTreat.filterNot(ctt => speedLimitsAlreadyExistents.exists(_.linkId == ctt.newId.get))
+        println("\nNumber of Changes to Treat: "+ changesWithoutSpeedLimitCreated)
+
+        changesWithoutSpeedLimitCreated.foreach { cws =>
+          val adjacents = roadLinkService.getAdjacent(cws.newId.getOrElse(0L), false).map(_.linkId).toSet
+          if (adjacents.size >= 2) {
+            val speedLimitsOnAdjacents = speedLimitDao.getCurrentSpeedLimitsByLinkIds(Some(adjacents))
+            if (speedLimitsOnAdjacents.size >= 2) {
+              val speedLimitsToCreate = speedLimitService.newChangeAsset(filteredRoadLinks, speedLimitsOnAdjacents, Seq(cws))
```

Comment:
You can use the val roadLinksToTreat in my previous suggestion to replace the filteredRoadLinks. It will contain less roadLinks so the find method inside this method will take less time

Reply:
Sorry this should not be applied